### PR TITLE
Add extra logging output and check for unexpected status codes

### LINKED ISSUE #4358: scripts/get-external-data.py provides little user feedback
### Expected behavior
Some idea that it is doing something

### Actual behavior
Some idea that it is doing something

### Links and screenshots illustrating the problem
renderaccount@ubuntuvm66:~/src/openstreetmap-carto$ time scripts/get-external-data.py
INFO:root:Checking table simplified_water_polygons
INFO:root:Checking table water_polygons
(that's been sat like that for about 10 minutes)

At least optionally "get-external-data.py" should provide feedback on what it is doing and how far it has got through the process.

### PR DESCRIPTION
**Pull request by jgruca**
Fixes #4358 

Changes proposed in this pull request:
- Add more verbose logging
- Explicitly check for status 200/304/other

New output (new data to download):
```
INFO:root:Starting load of external data into database
INFO:root:Checking table simplified_water_polygons
INFO:root:  Fetching https://osmdata.openstreetmap.de/download/simplified-water-polygons-split-3857.zip
INFO:root:  Download complete (23608777 bytes)
INFO:root:  File decompressed
INFO:root:  Importing into database
INFO:root:  Import complete
INFO:root:Checking table water_polygons
INFO:root:  Fetching https://osmdata.openstreetmap.de/download/water-polygons-split-3857.zip
INFO:root:  Download complete (781453667 bytes)
INFO:root:  File decompressed
INFO:root:  Importing into database
INFO:root:  Import complete
INFO:root:Checking table icesheet_polygons
INFO:root:  Fetching https://osmdata.openstreetmap.de/download/antarctica-icesheet-polygons-3857.zip
INFO:root:  Download complete (53216874 bytes)
INFO:root:  File decompressed
INFO:root:  Importing into database
INFO:root:  Import complete
INFO:root:Checking table icesheet_outlines
INFO:root:  Fetching https://osmdata.openstreetmap.de/download/antarctica-icesheet-outlines-3857.zip
INFO:root:  Download complete (53842470 bytes)
INFO:root:  File decompressed
INFO:root:  Importing into database
INFO:root:  Import complete
INFO:root:Checking table ne_110m_admin_0_boundary_lines_land
INFO:root:  Fetching https://naturalearth.s3.amazonaws.com/110m_cultural/ne_110m_admin_0_boundary_lines_land.zip
INFO:root:  Download complete (45820 bytes)
INFO:root:  File decompressed
INFO:root:  Importing into database
INFO:root:  Import complete
```

New output (data not updated):
```
INFO:root:Starting load of external data into database
INFO:root:Checking table simplified_water_polygons
INFO:root:  Fetching https://osmdata.openstreetmap.de/download/simplified-water-polygons-split-3857.zip
INFO:root:  Table simplified_water_polygons did not require updating
INFO:root:Checking table water_polygons
INFO:root:  Fetching https://osmdata.openstreetmap.de/download/water-polygons-split-3857.zip
INFO:root:  Table water_polygons did not require updating
INFO:root:Checking table icesheet_polygons
INFO:root:  Fetching https://osmdata.openstreetmap.de/download/antarctica-icesheet-polygons-3857.zip
INFO:root:  Table icesheet_polygons did not require updating
INFO:root:Checking table icesheet_outlines
INFO:root:  Fetching https://osmdata.openstreetmap.de/download/antarctica-icesheet-outlines-3857.zip
INFO:root:  Table icesheet_outlines did not require updating
INFO:root:Checking table ne_110m_admin_0_boundary_lines_land
INFO:root:  Fetching https://naturalearth.s3.amazonaws.com/110m_cultural/ne_110m_admin_0_boundary_lines_land.zip
INFO:root:  Table ne_110m_admin_0_boundary_lines_land did not require updating
```

Original output:
```
INFO:root:Checking table simplified_water_polygons
INFO:root:Table simplified_water_polygons did not require updating
INFO:root:Checking table water_polygons
INFO:root:Table water_polygons did not require updating
INFO:root:Checking table icesheet_polygons
INFO:root:Table icesheet_polygons did not require updating
INFO:root:Checking table icesheet_outlines
INFO:root:Table icesheet_outlines did not require updating
INFO:root:Checking table ne_110m_admin_0_boundary_lines_land
INFO:root:Table ne_110m_admin_0_boundary_lines_land did not require updating
```

### CODE REVIEW

File: scripts/get-external-data.py [hunk 1]
```diff
@@ -229,18 +231,23 @@ def main():
...
                     else:
                         new_last_modified = None
+
+                    logging.info("  Download complete ({} bytes)".format(len(download.content)))
+
                     if "archive" in source and source["archive"]["format"] == "zip":
                         zip = zipfile.ZipFile(io.BytesIO(download.content))
                         for member in source["archive"]["files"]:
                             zip.extract(member, workingdir)
+                        logging.info("  File decompressed")
```
Comment (pitdicker) [inline scripts/get-external-data.py:250]: I would change this to before the action takes place, with "Decompressing file"

**Comment by SomeoneElseOSM** [comment]
Does this give the same level of feedback as the version of the shapefile download script as of https://github.com/gravitystorm/openstreetmap-carto/releases/tag/v5.1.0 and earlier, like this:

    3-1. Downloading 'water-polygons-split-3857'...
      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                     Dload  Upload   Total   Spent    Left  Speed
      7  745M    7 55.1M    0     0   222k      0  0:57:14  0:04:14  0:53:00  283k

If so, great!  If not - how hard would it be just to throw away "get-external-data.py" and "external-data.yml" and just use the old "get-shapefiles.py" instead?

**Comment by jgruca** [comment]
The old get-shapefiles.py script called curl, which handled displaying progress and speed, among other things. However, it looks like it did not handle a lot of other things that get-external-data.py does, such as database updates. (See the commit message on https://github.com/gravitystorm/openstreetmap-carto/commit/67824003e02b27574de9b6f49313043248d5bcff for some details.)

We could switch over to using curl, similar to the old script, but there may be reasons why they moved away from curl to the requests library.

**Comment by SomeoneElseOSM** [comment]
Just for completeness, the script does still work as expected (I can render tiles using the results).  One more "nice to have" would be a timestamp on each line, and perhaps a first-line warning that some of the downloads are large and may take some time.

**Pull request merged**